src/main/java/org/reactome/addlinks/dataretrieval/pharos/PharosLigandDataRetriever.java

```diff
@@ -27,35 +27,107 @@
+	//          "name": "acetazolamide",
+	//          "isdrug": true,
+	//          "synonyms": [
+	//            {
+	//              "name": "unii",
+	//              "value": "O3FX965V0I"
+	//            },
+	//            {
+	//              "name": "PubChem",
+	//              "value": "1986"
+	//            },
+	//            {
+	//              "name": "Guide to Pharmacology",
+	//              "value": "6792"
+	//            },
+	//            {
+	//              "name": "ChEMBL",
+	//              "value": "CHEMBL20"
+	//            },
+	//            {
+	//              "name": "DrugCentral",
+	//              "value": "56"
+	//            },
+	//            {
+	//              "name": "pt",
+	//              "value": "ACETAZOLAMIDE"
+	//            },
+	//            {
+	//              "name": "LyCHI",
+	//              "value": "9XY1117CMPQQ"
+	//            }
+	//          ]
+	//        },
+	//        {
+	//          "ligid": "4UXS9UGRZQTR",
+	//          "name": "staurosporin",
+	//          "isdrug": false,
+	//          "synonyms": [
+	//            {
+	//              "name": "PubChem",
+	//              "value": "44259"
+	//            },
+	//            {
+	//              "name": "Guide to Pharmacology",
+	//              "value": "346"
+	//            },
+	//            {
+	//              "name": "ChEMBL",
+	//              "value": "CHEMBL388978"
+	//            },
+	//            {
+	//              "name": "LyCHI",
+	//              "value": "4UXS9UGRZQTR"
+	//            }
+	//          ]
+	//        },
+	//        ...
+
 	@Override
-	protected int processJSONArray(FileWriter fw, JSONObject jsonObj) throws IOException
+	protected int processJSONArray(FileWriter writer, JSONObject jsonObj) throws IOException
 	{
+		int i = 0;
 		try
 		{
 			JSONArray ligands = (((JSONObject)((JSONObject) jsonObj.get("data")).get("ligands")).getJSONArray("ligands"));
-			for (int i = 0; i < ligands.length(); i++)
+			for (i = 0; i < ligands.length(); i++)
 			{
 				JSONObject ligand = ((JSONObject)ligands.get(i));
-				Boolean isDrug = ligand.getBoolean("isdrug");
+				boolean isDrug = ligand.getBoolean("isdrug");
 				// Check that the ligand has "isdrug==true" and then get the GtP synonym's value (synonym's "name" will be "Guide to Pharmacology").
-				if (isDrug.booleanValue())
+				if (isDrug)
 				{
 					// Get the ligand ID used by Pharos
 					String ligid = ligand.getString("ligid");
 					String gtpIdentifier = null;
 					// Now we process the synonyms...
 					JSONArray synonyms = ligand.getJSONArray("synonyms");
-					for (int j = 0; j < synonyms.length(); j++)
+					boolean done = false;
+					int j = 0;
+					while (!done)
 					{
 						JSONObject synonym = (JSONObject)synonyms.get(j);
 						if (synonym.getString("name").equals("Guide to Pharmacology"))
 						{
 							gtpIdentifier = synonym.getString("value");
+							done = true;
 						}
+						j++;
+						done = j >= synonyms.length();
```

Comment:
I think this will still run for the full loop even after finding the gtpIdentifier because done gets reassigned on line 126 even if it was set to true on line 123 if the counter "j" is less than the synonym length.

It think it would be better to have the whole condition in the while loop itself instead of using `done` as an intermediate because you can say `while(gtpIdentifier == null || j < synonyms.length())` in one place.

Reply:
Good catch!